lib/liquid/block_body.rb

```diff
@@ -70,49 +70,57 @@ def render(context)
       output = []
       context.resource_limits.render_score += @nodelist.length
 
-      @nodelist.each do |token|
-        # Break out if we have any unhanded interrupts.
+      idx = 0
+      while node = @nodelist[idx]
+        render_node_to_output(node, output, context)
         break if context.interrupt?
```

Comment:
Hm. Not sure I understand why this `break` was at the beginning of the loop body before but now it's at the end of the loop body. Could you explain?

Reply:
If an interrupt happens during `render_node_to_output` it seems unnecessary to wait for the next iteration of the loop to break out of it.

Reply:
We could avoid this check entirely if `token.is_a?(String)` and we didn't extract code out into the `render_node_to_output` method.  Perhaps we could instead move the exception handling into `render_node` to simplify this method without losing the ability to call `break` inside the case statement.

Reply:
That's a good point. I'll try that.